resources/aws/rti_aws_uploader.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env python3.7
+#
+# (c) 2019 Copyright, Real-Time Innovations, Inc.  All rights reserved. RTI
+# grants Licensee a license to use, modify, compile, and create derivative
+# works of the Software.  Licensee has the right to distribute object form only
+# for use with RTI products.  The Software is provided "as is", with no
+# warranty of any type, including any warranty for fitness for any purpose. RTI
+# is under no obligation to maintain or support the Software.  RTI shall not be
+# liable for any incidental or consequential damages arising out of the use or
+# inability to use the software.
+"""Manage RTI ConnextDDS versions in AWS S3 Bucket.
+
+This module implements some functions to upload/delete/list RTI ConnextDDS
+versions from an AWS S3 bucket. If you have the credentials in a file located
+in a non-default location (~/.aws/config), you can specify it as an argument in
+the command line (-c <path_to_config>).
+
+Examples:
+    Upload dir to aws bucket:
+
+        python rti_aws_uploader.py -b bucket_name -d /home/user/directory
+
+    List items from aws bucket with custom credentials file location:
+
+        python rti_aws_uploader.py -b bucket_name -c ./folder/config -l
+
+    Delete folder in aws bucket:
+
+        python rti_aws_uploader.py -b bucket_name -r test_directory_name
+"""
+
+import os
+import sys
+import boto3
+import botocore
+import argparse
+
+from pathlib import Path
+from typing import List, Set
+
+
+def bucket_exists(s3, bucket_name: str):
+    """Checks if bucket exists.
+
+    Args:
+        s3 (boto3.resource('s3')): s3 resource to check the bucket from.
+        bucket_name (str): Selected bucket name to check.
+
+    Returns:
+        exists (bool): True if the bucket exists.
+        error (str): Error message
+    """
+    exists: bool = True
+    error: str = ''
+    try:
+        s3.meta.client.head_bucket(Bucket=bucket_name)
+    except botocore.client.ClientError as e:
+        if e.response['Error']['Code'] == '404':
+            exists = False
+            error = f'Bucket {bucket_name} not found.'
+    except botocore.exceptions.NoCredentialsError:
+        exists = False
+        error = 'Credentials not found.'
+    return exists, error
```

Comment:
Returning an error message to handle it is not a good practice. It would be better to create your own `Exception` class and raise an object of this new class.